proto/backend.proto

```diff
@@ -413,17 +396,63 @@ message MutationResponse {
   bytes object_bytes = 4;
 }
 
+//-----------------------------------------------------------------
+// ResourceConversion -- EXPERIMENTAL and is subject to change until 12.??
+//-----------------------------------------------------------------
+
+// ResourceConversion is a service that can be used to convert resources between versions
+// See: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-request-and-response
+service ResourceConversion {
+  // Convert objects to a target version
+  rpc ConvertObjects(ConversionRequest) returns (ConversionResponse);
+}
+
+// GroupVersion represents the API group and version of a resource.
+message GroupVersion{
+  string group = 1;
+  string version = 2;
+}
+
+// RawObject contains a resource serialized into a byte array with a content type
+message RawObject {
+  // Raw is the serialized object
+  bytes raw = 1;
+  // ContentType is the media type of the raw object
+  string content_type = 2;
+}
+
+// ConversionRequest supports converting objects from one version to another 
+message ConversionRequest {
+  // NOTE: this may not include app or datasource instance settings depending on the request
+  PluginContext pluginContext = 1;
+
+  // uid is an identifier for the individual request/response. It allows distinguishing instances of requests which are
+  // otherwise identical (parallel requests, etc).
+  // The UID is meant to track the round trip (request/response) between the Kubernetes API server and the webhook, not the user request.
+  // It is suitable for correlating log entries between the webhook and apiserver, for either auditing or debugging.
+  string uid = 2;
```

Comment:
`uid` is also something that is present in CRD conversion webhooks, so this is forwarding that information in case it's useful for debugging